### DISCUSSION TITLE
Carbon Aware SDK - Call for Features

### DISCUSSION THREAD

**Discussion by Sophietn** [Green-Software-Foundation/carbon-aware-sdk#521] [General]
## Call for Features

The Carbon Aware SDK has just released [v1.4](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/518)! 

We're now looking ahead, and planning out our future roadmap. 
And we want to hear from those using the SDK. 

### What do YOU want next for the Carbon Aware SDK? 

- Any features you'd love to see added?
- Any pain points you'd love to see fixed?

Leave your thoughts and ideas as comments below! 
Once your idea is fully formed,[ Raise a Feature Contribution via our issue template. 
](https://github.com/Green-Software-Foundation/carbon-aware-sdk/issues/new?assignees=vaughanknight&labels=feature%2Ctriage&projects=&template=contribution-issue.yml&title=%5BFeature+Contribution%5D%3A+)

This discussion is a Call for Features. 
Whilst there's no guarantee your requests will be implemented, we want to gather feedback and hear directly from SDK users as to what you'd most like to see next.

**Comment by YaSuenag** [comment]
I'd like to propose Multiple data source. I believe this feature helps to operate onpremise and/or multi AZ system.

I proposed this feature in #345 , but it was closed as "stale" unfortunately. So I'm happy if we can raise this proposal again.

**Comment by bderusha** [comment]
I propose adding "Publishing the SDK as a NuGet package" be included on the roadmap.

There are a few things I would like to see happen as part of such work:

- Review + Merge the ADR-0006 implementation https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/366
- Update ADR-0014 to include a detailed proposal for how to split the project into packages
- Implementation of ADR-0014 to structure the code to support packaging
- Implementation of package publishing pipelines

Benefits:
- Creating a published "SDK package" opens up the SDK to a wider audience by making it easier to integrate into existing software, thereby increasing the impact of the project.
- Creating a "core package" allows community members to create their own DataSources without needing to clone the entire project.
